pkg/ccl/testccl/sqlccl/sqlccl_test: TestExplainGist failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainGist [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17437809?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17437809?buildTab=artifacts#/) on master @ [52e38aafd852b646456000a04e1dc440627d2c6d](https://github.com/cockroachdb/cockroach/commits/52e38aafd852b646456000a04e1dc440627d2c6d):


```
=== RUN   TestExplainGist
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/17d4c94e97ab6a70b4ba9df15e5f218d/logTestExplainGist169795328
    test_log_scope.go:76: use -show-logs to present logs inline
    explain_test.go:308: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/17d4c94e97ab6a70b4ba9df15e5f218d/logTestExplainGist169795328
--- FAIL: TestExplainGist (12.16s)
=== RUN   TestExplainGist/main
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    explain_test.go:255: pq: query execution canceled due to statement timeout
    --- FAIL: TestExplainGist/main (10.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132828 pkg/ccl/testccl/sqlccl/sqlccl_test: TestExplainGist failed [C-test-failure O-robot T-sql-queries branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainGist.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43550

--- a/pkg/ccl/testccl/sqlccl/explain_test.go
+++ b/pkg/ccl/testccl/sqlccl/explain_test.go
@@ -249,9 +249,11 @@ func TestExplainGist(t *testing.T) {
 			}
 			_, err = sqlDB.Exec("SELECT crdb_internal.decode_plan_gist($1);", gist)
 			if err != nil {
-				// We might be still in the process of cancelling the previous
-				// DROP operation - ignore this particular error.
-				if !testutils.IsError(err, "descriptor is being dropped") {
+				// We might be still in the process of cancelling the previous DROP
+				// operation or hit the statement timeout - ignore this particular
+				// error.
+				if !testutils.IsError(err, "descriptor is being dropped") &&
+					!sqltestutils.IsClientSideQueryCanceledErr(err) {
 					t.Fatal(err)
 				}
 				continue

